taxi/views.py

```diff
@@ -87,3 +105,15 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+@login_required
+def assign_car(request: HttpRequest, pk: int) -> HttpResponse:
+    car = Car.objects.get(pk=pk)
+
+    if request.user in car.drivers.all():
+        car.drivers.remove(request.user)
+    else:
+        car.drivers.add(request.user)
```

Comment:
use .exists() so that db uses WHERE and EXISTS at the same time because otherwise you firstly fetch 1 000 000 records and then python loops through all records to check "in" condition